internal/config/config.go

```diff
@@ -27,15 +28,18 @@ type LoggingConfig struct {
 type ApiConfig struct {
 	ListenAddress  string `yaml:"address" envconfig:"API_LISTEN_ADDRESS"`
 	ListenPort     uint   `yaml:"port" envconfig:"API_LISTEN_PORT"`
-	MetricsAddress string `yaml:"metricsAddress" envconfig:"METRICS_LISTEN_ADDRESS"`
-	MetricsPort    uint   `yaml:"metricsPort" envconfig:"METRICS_LISTEN_PORT"`
 }
 
 type DebugConfig struct {
 	ListenAddress string `yaml:"address" envconfig:"DEBUG_ADDRESS"`
 	ListenPort    uint   `yaml:"port" envconfig:"DEBUG_PORT"`
 }
 
+type MetricsConfig struct {
+	MetricsAddress string `yaml:"metricsAddress" envconfig:"METRICS_LISTEN_ADDRESS"`
+	MetricsPort    uint   `yaml:"metricsPort" envconfig:"METRICS_LISTEN_PORT"`
```

Comment:
The YAML keys should just be `address` and `port` with it under its own top-level key. The vars should also be renamed to `ListenAddress` and `ListenPort` for consistency with `api` and `debug`.